### DISCUSSION TITLE
How to store env var?

### DISCUSSION THREAD

**Discussion by udondan** [johnlindquist/kit#230] [Q&A] [answered]
I know there is `await env('foo')`, which let's me ask the user for an env var **once**, that will then be stored the `.kenv/.env` file.

How can I store a value regardless if it is already present in the .env file? Of course I could just do some file manipulation but I was wondering if there is a method available that can do this.

I also could just use my own file to store. But that would probably be not ideal. I was looking at [Davids AWS log group viewer](https://github.com/johnlindquist/kit/discussions/199) and plan to write a couple of similar scripts for my daily routines. Davids script stores the `AWS_PROFILE`. But here's a the catch. I work with a lot of different AWS accounts and profiles and I need a way to handle profile switches. But at the same time I do not _always_ want to fill in the profile when I run a script.

If the community could agree on using the var `AWS_PROFILE` to store that info, a separate profile switcher script would be a central component. So that's what I'm looking to solve here.

**Comment by johnlindquist** [comment, marked as answer]
`await env("foo", {reset:true})` should prompt for the var every time and overwrite it.

**Reply by udondan** [reply]
Interesting. This works! How though? I cannot find the string `reset` anywhere in the project. 😺 

I just found the `set-env-var` function. I'll be using this, because I want to fill the value programmatically and not ask the user for input.

```js
await cli('set-env-var', 'foo', 'bar');
```

**Reply by johnlindquist** [reply]
"reset" is in the beta branch. I'm guessing you searched on main

**Comment by udondan** [comment]
I published the script here via #232
